### PR TITLE
[TTAHUB-828] Activity summary duration input validation

### PR DESCRIPTION
**Pull request by nvms**
## Description of change

- Define `max` prop on duration input so that keyboard uparrow input prevents values of >99.5.
- Input value >= 100 shows a validation error message that reads: "Duration must be less than 100 hours".
- Add a test for two duration validation errors: 1) negative values and 2) >= 100 values

## How to test

In the duration field of the activity summary form:

- When incrementing by either scrolling with the mousewheel or using the uparrow keys, max value should be clamped to 99.5:

https://user-images.githubusercontent.com/17074357/183146487-fba03025-abcb-4f16-aa87-7bb617d345a8.mov

- When input is by another means set to a value > 99.5, a helpful validation error message should be visible:

https://user-images.githubusercontent.com/17074357/183146686-f021c332-77e0-44a1-b5fb-9149ca208310.mov

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-828

### Every PR

- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/activitySummary.js [hunk 1]
```diff
@@ -296,13 +296,15 @@ const ActivitySummary = ({
...
                   min={0}
+                  max={99.5}
                   step={0.5}
                   inputRef={
                     register({
                       required: 'Please enter the duration of the activity',
                       valueAsNumber: true,
                       pattern: { value: /^\d+(\.[0,5]{1})?$/, message: 'Duration must be rounded to the nearest half hour' },
                       min: { value: 0, message: 'Duration can not be negative' },
+                      max: { value: 99.5, message: 'Duration must be less than 100 hours' },
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/Pages/activitySummary.js:307]: Do we want them to be able to enter a value of 0 for duration?

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good. :smile:

I would just follow up with Angela if we want to allow them to have a value of 0 for duration.

**Review by thewatermethod — APPROVED** [review]
🔥

**Comment by nvms** [comment]
Confirmed with Angela the acceptable range is 0.5-99.
